plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1450,6 +1488,63 @@ private List<ColumnMetadata> getColumnMetadatas(Schema schema)
         return columns.build();
     }
 
+    private ConnectorTableMetadata createConnectorMetadataWithExistingMetadata(TrinoFileSystem fileSystem, ConnectorTableMetadata tableMetadata, String metadataLocation)
+    {
+        FileIO fileIO = fileSystem.toFileIo();
+        InputFile metadataFile = fileIO.newInputFile(metadataLocation);
+        TableMetadata existingMetadata = TableMetadataParser.read(fileIO, metadataFile);
+
+        Map<String, Object> existingTableMetadataProps = new HashMap<>();
+        existingTableMetadataProps.putAll(tableMetadata.getProperties());
+        for (Map.Entry<String, String> entry : existingMetadata.properties().entrySet()) {
+            existingTableMetadataProps.put(entry.getKey(), (Object) entry.getValue());
+        }
+        existingTableMetadataProps.put(EXISTING_LATEST_METADATA_LOCATION, metadataLocation);
+
+        Optional<String> comment = Optional.ofNullable((String) existingTableMetadataProps.get(TABLE_COMMENT));
+        return new ConnectorTableMetadata(tableMetadata.getTable(), getColumnMetadatas(existingMetadata.schema()), existingTableMetadataProps, comment);
+    }
+
+    private static Optional<String> getLatestMetadataLocation(TrinoFileSystem fileSystem, String location)
+    {
+        String latestMetadataLocation = null;
+        try {
+            if (fileSystem.listFiles(location).hasNext()) {
+                String metadataDirectoryLocation = stripTrailingSlash(location) + File.separator + METADATA_FOLDER_NAME;
+                FileIterator fileIterator = fileSystem.listFiles(metadataDirectoryLocation);
+                int latestMetadataVersion = -1;
+                while (fileIterator.hasNext()) {
+                    FileEntry fileEntry = fileIterator.next();
+                    if (fileEntry.path().endsWith(METADATA_FILE_EXTENSION)) {
+                        int version = parseVersion(fileEntry.path());
+                        if (version > latestMetadataVersion) {
```

Comment:
It is possible that there could be a tie here. For example, if two writers tried updating the table at the same time. If we detect that there is a tie for the highest version number, we should throw an exception.

Reply:
Will take care in `register_table` procedure.